### ISSUE TITLE
Improving the inferred type of an optional property

### ISSUE THREAD

**Issue by Thanaen** [fabian-hiller/valibot#9]
Take this object, created with zod:
```ts
const ZodProblemDetailsSchema = z.object({
  detail: z.string().optional(),
  status: z.number().optional(),
  title: z.string().optional(),
  type: z.string().optional(),
});
```

When we infer the type of this schema with the z.infer method, we see that all properties are optional
```ts
type ZodProblemDetails = {
    detail?: string | undefined;
    status?: number | undefined;
    title?: string | undefined;
    type?: string | undefined;
}
```

I've implemented the same thing with valibot:
```ts
const ProblemDetailsSchema = object({
  detail: optional(string()),
  status: optional(number()),
  title: optional(string()),
  type: optional(string()),
});
```

The type inferred with Output gives this: 
```ts
type ProblemDetails = {
    detail: string | undefined;
    status: number | undefined;
    title: string | undefined;
    type: string | undefined;
}
```

Would it be possible, as zod does, to mark optional properties of an object as optional, and not just possibly undefined? 

(Nice work on this library, by the way!)

**Comment by fabian-hiller** [comment]
Thanks for the tip. I will have a look at it.

**Comment by fabian-hiller** [comment]
Should be fixed in v0.8.0.
